Uplift/Views/GymDetail/GymDetailHoursCell.swift

```diff
@@ -0,0 +1,253 @@
+            return "\(dailyGymHours.openTime.getStringOfDatetime(format: "h:mm a")) - \(dailyGymHours.closeTime.getStringOfDatetime(format: "h:mm a"))"
+        } else {
+            return "Closed"
+        }
+    }
+
+    // MARK: - Targets
+    @objc func dropHours(sender: UITapGestureRecognizer) {
+        hoursTableView.beginUpdates()
+        var modifiedIndices: [IndexPath] = []
+        for i in 0..<6 {
+            modifiedIndices.append(IndexPath(row: i, section: 0))
+        }
+
+        if GymDetailHoursCell.hoursData.isDropped { // collapsing details
+            GymDetailHoursCell.hoursData.isDropped = false
+            hoursTableView.deleteRows(at: modifiedIndices, with: .fade)
+            // swiftlint:disable:next force_cast
+            (hoursTableView.headerView(forSection: 0) as! GymHoursHeaderView).downArrow.image = .none
+            // swiftlint:disable:next force_cast
+            (hoursTableView.headerView(forSection: 0) as! GymHoursHeaderView).rightArrow.image = UIImage(named: "right-arrow-solid")
+
+            self.delegate?.didDropHours(isDropped: false) { () in
+                UIView.animate(withDuration: 0.3) {
+                    self.setupConstraints()
+                    self.contentView.layoutIfNeeded()
+                }
+            }
+        } else { // expanding details
+            GymDetailHoursCell.hoursData.isDropped = true
+            hoursTableView.insertRows(at: modifiedIndices, with: .fade)
+            // swiftlint:disable:next force_cast
+            (hoursTableView.headerView(forSection: 0) as! GymHoursHeaderView).downArrow.image = UIImage(named: "down-arrow-solid")
+            // swiftlint:disable:next force_cast
+            (hoursTableView.headerView(forSection: 0) as! GymHoursHeaderView).rightArrow.image = .none
+
+            self.delegate?.didDropHours(isDropped: true) { () in
+                UIView.animate(withDuration: 0.5) {
+                    self.setupConstraints()
+                    self.contentView.layoutIfNeeded()
+                }
+            }
+        }
+        hoursTableView.endUpdates()
+    }
+}
+
+extension GymDetailHoursCell: UITableViewDelegate, UITableViewDataSource {
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        let numRows = GymDetailHoursCell.hoursData.isDropped ? 6 : 0
+        return numRows
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        // swiftlint:disable:next force_cast
+        let cell = tableView.dequeueReusableCell(withIdentifier: GymHoursCell.identifier, for: indexPath) as! GymHoursCell
+
+        let date = Date()
+        let day = (date.getIntegerDayOfWeekToday() + indexPath.row + 1) % 7
+        cell.hoursLabel.text = getStringFromDailyHours(dailyGymHours: hours[day])
+
+        switch days[day] {
+        case .sunday:
+            cell.dayLabel.text = "Su"
```

Comment:
we use these abbreviations for days in `OldGymDetailViewController` too so lets just move them. into a global constants file